Refactor extended classes

ocombe opened the pull request:
## PR Type
What kind of change does this PR introduce?
```
[x] Refactoring (no functional changes, no api changes)
```

## What is the current behavior?
Somes classes are duplicated with one abstract class and its actual implementation for historical reasons that are no longer needed.

## What is the new behavior?
Duplicated classes have been merged when possible

## Does this PR introduce a breaking change?
```
[x] No
```

## Other information
These commits change the API guard files, but they just remove the "abstract" keywords and add the constructors, the api themselves don't change.
If you still consider this an API change, I should probably update the commits to change "refactor" to "fix"

mary-poppins commented:
You can preview 9547c45 at https://pr19143-9547c45.ngbuilds.io/.

mhevery commented on packages/core/src/application_module.ts:
```diff
@@ -35,9 +40,14 @@ export function _localeFactory(locale?: string): string {
-    ApplicationRef_,
-    {provide: ApplicationRef, useExisting: ApplicationRef_},
     ApplicationInitStatus,
+    {
```
I think you can just say `ApplicationRef` I don't think the rest is necessary.

ocombe replied:
That's what I did first, but for some reason it doesn't work because it says that it cannot resolve all of the dependencies

mhevery commented on tools/public_api_guard/core/index.d.ts:
```diff
@@ -125,15 +125,17 @@ export declare class ApplicationModule {
+    readonly components: ComponentRef<any>[];
+    readonly isStable: Observable<boolean>;
+    readonly viewCount: number;
+    constructor(_zone: NgZone, _console: Console, _injector: Injector, _exceptionHandler: ErrorHandler, _componentFactoryResolver: ComponentFactoryResolver, _initStatus: ApplicationInitStatus);
```
Can you mark the constructor as `/** @internal */` so that it does not show up here.

mhevery commented on tools/public_api_guard/core/index.d.ts:
```diff
@@ -695,13 +697,14 @@ export declare const PLATFORM_INITIALIZER: InjectionToken<(() => void)[]>;
+export declare class PlatformRef {
+    readonly destroyed: boolean;
+    readonly injector: Injector;
+    constructor(_injector: Injector);
```
`/** @internal */`

mhevery commented on tools/public_api_guard/http/index.d.ts:
```diff
@@ -79,15 +79,18 @@ export declare class Jsonp extends Http {
 /** @deprecated */
-export declare abstract class JSONPBackend extends ConnectionBackend {
+export declare class JSONPBackend extends ConnectionBackend {
+    constructor(_browserJSONP: BrowserJsonp, _baseResponseOptions: ResponseOptions);
```
`/** @internal */`

mhevery commented on tools/public_api_guard/http/index.d.ts:
```diff
@@ -79,15 +79,18 @@ export declare class Jsonp extends Http {
     request: Request;
     response: Observable<Response>;
-    abstract finished(data?: any): void;
+    constructor(req: Request, _dom: BrowserJsonp, baseResponseOptions?: ResponseOptions);
```
`/** @internal */`

mhevery commented on tools/public_api_guard/platform-webworker/index.d.ts:
```diff
@@ -2,13 +2,16 @@
-    abstract runOnService(args: UiArguments, returnType: Type<any> | SerializerTypes | null): Promise<any> | null;
+export declare class ClientMessageBroker {
+    channel: any;
+    constructor(messageBus: MessageBus, _serializer: Serializer, channel: any);
```
`/** @internal */`

mhevery commented on tools/public_api_guard/platform-webworker/index.d.ts:
```diff
@@ -2,13 +2,16 @@
-export declare abstract class ClientMessageBrokerFactory {
-    abstract createMessageBroker(channel: string, runInZone?: boolean): ClientMessageBroker;
+export declare class ClientMessageBrokerFactory {
+    constructor(_messageBus: MessageBus, _serializer: Serializer);
```
`/** @internal */`

mhevery commented on tools/public_api_guard/platform-webworker/index.d.ts:
```diff
@@ -62,13 +65,16 @@ export declare const enum SerializerTypes {
-    abstract registerMethod(methodName: string, signature: Array<Type<any> | SerializerTypes> | null, method: Function, returnType?: Type<any> | SerializerTypes): void;
+export declare class ServiceMessageBroker {
+    channel: string;
+    constructor(messageBus: MessageBus, _serializer: Serializer, channel: string);
```
`/** @internal */`

mhevery commented on tools/public_api_guard/platform-webworker/index.d.ts:
```diff
@@ -62,13 +65,16 @@ export declare const enum SerializerTypes {
-export declare abstract class ServiceMessageBrokerFactory {
-    abstract createMessageBroker(channel: string, runInZone?: boolean): ServiceMessageBroker;
+export declare class ServiceMessageBrokerFactory {
+    constructor(_messageBus: MessageBus, _serializer: Serializer);
```
`/** @internal */`

mhevery commented:
Also could you run `./build.sh` before and after your change and include the files sizes changes of the `.umd.js` files in the commit message? This way we have some ideas about savings.

mary-poppins commented:
You can preview 13f3e37 at https://pr19143-13f3e37.ngbuilds.io/.

ocombe commented:
| package | before | after |
| -------| ------|---------|
| core | 496 895 bytes | 498 950 bytes |
| http |  78 827 bytes | 78 407 bytes | 
| platform-webworker | 103 800 bytes | 102 630 bytes |

mary-poppins commented:
You can preview 8bbbe66 at https://pr19143-8bbbe66.ngbuilds.io/.

mhevery commented on tools/public_api_guard/http/index.d.ts:
```diff
@@ -79,15 +79,16 @@ export declare class Jsonp extends Http {
 /** @deprecated */
-export declare abstract class JSONPBackend extends ConnectionBackend {
+export declare class JSONPBackend extends ConnectionBackend {
+    createConnection(request: Request): JSONPConnection;
```
Looks like you made a new method part of public API. Could you remove it?

mhevery commented:
I think you can squash them into a single commit. Also include this table https://github.com/angular/angular/pull/19143#issuecomment-328641379 in the commit message of the SHA.

ocombe committed:
refactor: core, http & platform-webworker to remove public private class separation

The private classes `ApplicationRef_`, `PlatformRef_`, `JSONPConnection_`, `JSONPBackend_`, `ClientMessageBrokerFactory_`, `ServiceMessageBroker_`, `ClientMessageBroker_` and `ServiceMessageBrokerFactory_` have been removed and merged into their public equivalents.

The size of the minified umd bundles have been slightly decreased:
| package            | before     | after      |
| -------------------|------------|------------|
| core               | 217.791 kb | 217.144 kb |
| http               | 33.260 kb  | 32.838 kb  | 
| platform-webworker | 56.015 kb  | 54.933 kb  |

mary-poppins commented:
You can preview a62f78f at https://pr19143-a62f78f.ngbuilds.io/.

mary-poppins commented:
You can preview b9ec834 at https://pr19143-b9ec834.ngbuilds.io/.
